### PR TITLE
BIT-19: Add StartDatabaseFromScratch option to appsettings.json

### PR DESCRIPTION
**Pull request by dario-ramos**

### CODE REVIEW

File: bitprim.restapi/Startup.cs [hunk 1]
```diff
@@ -82,6 +69,27 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
             app.UseMvc();
         }
 
+        private void StartBitprimNode(IServiceCollection services)
+        {
+            // Initialize and register chain service
+            NodeConfig config = Configuration.Get<NodeConfig>();
+            exec_ = new Executor(config.NodeConfigFile, 0, 0);
```
Comment (fpelliccioni) [inline bitprim.restapi/Startup.cs]: Is there a way on C#/dotNet to get the stdout and stderr pointer/handlers?

File: bitprim.restapi/appsettings.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "BlockchainHeight" : 1260934,
```
Comment (fpelliccioni) [inline bitprim.restapi/appsettings.json]: Why do you store the blockchain height here?

Comment (dario-ramos) [reply bitprim.restapi/appsettings.json]: Suggested by Guillermo to compute sync status

File: bitprim.restapi/Startup.cs (hunk 1 above)
Comment (dario-ramos) [reply bitprim.restapi/Startup.cs]: Yes, on Windows for sure; haven't tested Linux

File: bitprim.restapi/Startup.cs [hunk 3]
```diff
@@ -82,6 +69,27 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
...
+            if(config.StartDatabaseFromScratch)
+            {
+                bool ok = exec_.InitChain();
+                if(!ok)
+                {
+                    throw new ApplicationException("Executor::InitChain failed; check log");
+                }
+            }
+            int result = exec_.RunWait();
+            if (result != 0)
```
Comment (mdallago) [inline bitprim.restapi/Startup.cs]: Following the framework design guidelines it's preferable not return an error code but throw  a custom subclass of Exception 
https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions

File: bitprim.restapi/Startup.cs (hunk 1 above)
Comment (mdallago) [reply bitprim.restapi/Startup.cs]: The last two parameters of Executor's constructor aren't very idiomatic.

File: bitprim.restapi/Startup.cs (hunk 3 above)
Comment (dario-ramos) [reply bitprim.restapi/Startup.cs]: I considered this case as a normal scenario, so used error checking just like the link suggests

**Pull request merged**